go/vt/vtgate/filters/filters.go

```diff
@@ -0,0 +1,16 @@
+package filters
+
+import (
+	"flag"
+
+	"vitess.io/vitess/go/flagutil"
+)
+
+// WatchKeyspaces - if provided this specifies which keyspaces should be
+// visible to a vtgate. By default the vtgate will allow access to any
+// keyspace.
+var WatchKeyspaces flagutil.StringListValue
+
+func init() {
+	flag.Var(&WatchKeyspaces, "keyspaces_to_watch", "Specifics which keyspaces this vtgate should have access to while routing queries or accessing the vschema")
```

Comment:
IMO this should be just in the top-level `vtgate.go` file alongside the other vtgate-wide flags.

Reply:
Agreed, but that won't fly because this is a `go/vt/vtgate` flag that needs visibility in `go/vt/vtgate/gateway` so we can enforce exclusivity. If we try to declare it there then I'll end up with an import cycle: `vtgate/vtgate.go -> gateway/discoverygateway.go -> vtgate/vtgate.go`

Options, as I saw them, were:

1. expose `keyspaces_to_watch` globally as I did allowing vtgate and gateway packages to pull it in
2. expose `tablet_filters` globally allowing ... etc

I chose option 1 because tablet_filters seemed more specialized to the discovery gateway than keyspaces_to_watch.

Would be excited if there is a 3rd option I just haven't learned about w/ go.